packages/loader/container-loader/src/containerContext.ts

```diff
@@ -169,6 +169,15 @@ export class ContainerContext implements IContainerContext {
 
     private readonly _fluidModuleP: Promise<IFluidModuleWithDetails>;
 
+    /**
+     * {@inheritDoc @fluidframework/container-definitions#IContainerContext.entryPoint}
+     */
+    public readonly entryPoint: Promise<FluidObject | undefined>
+    private _entryPointPromiseResolver: (entryPoint: FluidObject | undefined) => void
```

Comment:
look at the Deferred class in common-utils. it basically does this

Reply:
Got a version working locally with Deferred, thanks for the tip. However, I'll look into "disentangling" the flow with [Matt's suggestion](https://github.com/microsoft/FluidFramework/pull/12487#discussion_r1080692952).